File: src/AngularClient/src/app/core/services/product/search-products-request.ts
import { ProductSortOption } from './product-sort-option';

export interface SearchProductsRequest {
  pageIndex: number;
  pageSize: number;
  brandId?: number;
  categoryId?: number;
  name?: string;
  minPrice?: number;
  maxPrice?: number;
  sortBy?: ProductSortOption;
}

File: src/TechShelf.API/Requests/Products/SearchProductsRequest.cs
using System.ComponentModel.DataAnnotations;
using TechShelf.Domain.Products.Enums;

namespace TechShelf.API.Requests.Products;

public record SearchProductsRequest(
    [Range(1, int.MaxValue, ErrorMessage = "Page index must be greater than 0")] int PageIndex,
    [Range(1, int.MaxValue, ErrorMessage = "Page size must be greater than 0")] int PageSize,
    int? BrandId = null,
    int? CategoryId = null,
    string? Name = null,
    int? MinPrice = null,
    int? MaxPrice = null,
    ProductsSortOption? SortBy = null);
